src/viur/core/bones/phone.py

```diff
@@ -0,0 +1,119 @@
+import re
+import typing as t
+
+from viur.core.bones.string import StringBone
+from viur.core.bones.base import ReadFromClientError, ReadFromClientErrorSeverity
+
+DEFAULT_REGEX = r"^\+?(\d{1,3})[-\s]?(\d{1,4})[-\s]?(\d{1,4})[-\s]?(\d{1,9})$"
+
+
+class PhoneBone(StringBone):
+    """
+    The PhoneBone class is designed to store validated phone/fax numbers in configurable formats.
+    This class provides a number validation method, ensuring that the given phone/fax number conforms to the
+    required/configured format and structure.
+    """
+
+    type: str = "str.phone"
+    """
+    A string representing the type of the bone, in this case "str.phone".
+    """
+
+    def __init__(
+        self,
+        *,
+        test: t.Optional[t.Pattern[str]] = DEFAULT_REGEX,
+        max_length: int = 15,  # maximum allowed numbers according to ITU-T E.164
+        default_country_code: t.Optional[str] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        """
+        Initializes the PhoneBone with an optional custom regex for phone number validation, a default country code,
+        and a flag to apply the default country code if none is provided.
+
+        :param test: An optional custom regex pattern for phone number validation.
+        :param max_length: The maximum length of the phone number. Passed to 'StringBone'.
+        :param default_country_code: The default country code to apply.
+        If None is provided the PhoneBone will ignore auto prefixing of the country code.
+        :param kwargs: Additional keyword arguments. Passed to 'StringBone'.
+        """
+        self.test: t.Pattern[str] = re.compile(test) if isinstance(test, str) else test
+        self.default_country_code: t.Optional[str] = default_country_code
+        super().__init__(max_length=max_length, **kwargs)
+
+    @staticmethod
+    def _extract_digits(value: str) -> str:
+        """
+        Extracts and returns only the digits from the given value.
+
+        :param value: The input string from which to extract digits.
+        :return: A string containing only the digits from the input value.
+        """
+        return re.sub(r"[^\d+]", "", value)
+
+    def isInvalid(self, value: str) -> t.Optional[str]:
+        """
+        Checks if the provided phone number is valid or not.
+
+        :param value: The phone number to be validated.
+        :return: An error message if the phone number is invalid or None if it is valid.
+
+        The method checks if the provided phone number is valid according to the following criteria:
+        1. The phone number must not be empty.
+        2. The phone number must match the provided or default phone number format.
+        3. The phone number cannot exceed 15 digits, or the specified maximum length if provided (digits only).
+        """
+        if not value:
+            return "No value entered"
+
+        if not self.test.match(value):
```

Comment:
What if self.test is None?
```suggestion
        if self.test and not self.test.match(value):
```